platform-sdk/docs/proposals/metric-labels/README.md

```diff
@@ -0,0 +1,343 @@
+Here we need to check that a platform metric (that has a `nodeId` label) is still working correctly and ends in the exact same output as today.
+
+## Alternatives
+
+Since the idea of this proposal is to make the metrics module independent of the platform module we need to remove
+the usage of the `NodeId` class or extract the class from the platform and define it as general API.
+Since the class does not make any sense outside the consensus node (like in block node for example) this is not a good idea.
+Another option would be to change the type of the `nodeId` to a `String` but that would end in an ugly API for any
+system next to the platform that uses the metrics module since `null` or any other default must always be used as the `nodeId`.
+Next to that all that ideas will end in having a hardcoded `nodeId` label in the metrics that will be forwarded to the monitoring system.
+
+### Definition of Label at measurement time
+
+In the given concept the labels are defined when a metric is created.
+This is how several other libraries handle labels.
+Another option would be to define the labels at the time when a metric is measured.
+Both variants have their pros and cons.
+Since this proposal introduces a new concept to the metrics module we should start with the simplest solution.
+This does not mean that we cannot add the second option later by extending the public API.
+
+Let's have a look at the pros and cons of both options to understand the differences:
+
+The following sample creates a metric with labels at creation time:
+
+```java
+final Label transactionTypeLabel = new Label("transactionType", "fileUpload");
+final Counter.Config config = new Counter.Config("transactionsCategory", "transactionCount").withLabels(transactionTypeLabel);
+final Counter counter = metrics.getOrCreate(config);
+
+counter.increment();
+```
+
+Let's see how that metric can be used in a filter that filter transactions and counts the number of transactions of type `fileUpload`:
+
+```java
+class TransactionFilter {
+    
+    private final static Counter.Config config = new Counter.Config("transactionsCategory", "transactionCount");
+
+    private final Metrics metrics;
+    
+    TransactionFilter(Metrics metrics) {
+        this.metrics = metrics;
+    }
+
+    void handle(Transaction transaction) {
+        if(transaction.getType().equals("fileUploadTransaction")) {
+            Counter.Config configWithLabel = config.withLabels(new Label("transactionType", "fileUpload"));
+            Counter counter = metrics.getOrCreate(config);
+            counter.increment();
+        } else if(transaction.getType().equals("fileDeleteTransaction")) {
+            Counter.Config configWithLabel = config.withLabels(new Label("transactionType", "fileDelete"));
+            Counter counter = metrics.getOrCreate(config);
+            counter.increment();
+        }
```

Comment:
This is not a realistic example. We have ~50 transaction types, and the number is increasing. A method with 50 if-else branches will hopefully never make it through code review. 🙂

Reply:
ok, do you have a better one? I just wanted to have any example